### ISSUE TITLE
BUG: pam_cgroup simply doesn't work with cgroup2

### ISSUE THREAD

**Issue by 0n-s** [libcgroup/libcgroup#84]
pam_cgroup, at least under a cgroup2-only environment (maybe cgroup1 as well, I haven't tested), simply doesn't seem to work at all. It still acts as if it succeeded.

Tracing a random sshd instance for every file it opens, there are only 3:
```
25666 sshd                3   0 /proc/cgroups
25666 sshd                5   0 /proc/mounts
25666 sshd                7   0 /sys/fs/cgroup/cgroup.controllers
```
(nothing else cgroup or libcgroup-related)
For whatever reason cgrules.conf is never loaded. Despite this, it claims it changed the cgroup successfully (with `debug` flag):
```
Dec 26 13:49:12 h pam_cgroup(sshd:session): Changed cgroup for process 25666  with username [redacted].
```
This is not true, as we can see from the just logged in session:
```
[redacted]@h $ cat /proc/self/cgroup
0::/
```
This failure also doesn't result in the login failing, even though in my PAM configuration pam_cgroup is not optional but required.

(tested with libcgroup v2.0)

**Comment by drakenclimber** [comment]
Thanks for the bug report, @0n-s.

The pam module hasn't been a priority for the current users of libcgroup, so we have not emphasized development or testing in this area.  I'm uncertain how much the libcgroup/pam module is being used, as this is the first issue reported against it even though v2.0 has been out 7+ month.  Hmmmm....

**Comment by drakenclimber** [comment]
I'm going to assign this to v2.0.1.  I'm not certain if it will be fixed in that release, but I would like us to make a decision on how to proceed with pam by that release.

Thanks again for the report!

**Comment by lockywolf** [comment]
The pam module is a great thing to have. Cgrulesengd is a hack anyway.

I think the Slackware community has seen similar problems with cgroupv1 (and had to rely on cgrulesengd), but sometimes people are just too lazy or shy to report bugs :(.

**Comment by drakenclimber** [comment]
> The pam module is a great thing to have. Cgrulesengd is a hack anyway.

Won't argue with the cgrules comment :)

> I think the Slackware community has seen similar problems with cgroupv1 (and had to rely on cgrulesengd), but sometimes people are just too lazy or shy to report bugs :(.

That's good to know.  Thank you.

**Comment by drakenclimber** [comment]
Thanks for the comments, @0n-s and @lockywolf.  I have merged a fix into `main` and `release-2.0`.  Let me know if you have any issues.

I'm hoping to get out another release of the 2.0 branch in the next month or so.  This fix will obviously be in it.

Thanks again1
